solutionbox/structured_data/datalab_solutions/structured_data/trainer/task.py

```diff
@@ -145,62 +154,56 @@ def get_export_signature(examples, features, predictions):
     target_name = train_config['target_column']
     key_name = train_config['key_column']
 
-
     if util.is_classification_model(args.model_type):
+      
+      # Get the label of the input target.
       string_value = util.get_vocabulary(args.preprocess_output_dir, target_name)
-      outputs = {
-          key_name: tf.squeeze(features[key_name]).name,
-          'input_target_from_features': tf.squeeze(features[target_name]).name,
-      }
- 
-      print('8'*1000)
-      print('target tensor', features[target_name])
-      print('predictions', predictions)
-      # Try getting the target prediction prob.
-
-      # Get score of the target column.
       input_target_label = tf.contrib.lookup.index_to_string(
           features[target_name],
-          mapping=string_value)
-      input_target_score = vector_slice(predictions, features[target_name])
+          mapping=string_value)   
 
-      outputs.update({
-        'input_target_label': tf.squeeze(input_target_label).name,
-        'score_from_target_label': tf.squeeze(input_target_score).name,
-        'all_score': tf.squeeze(predictions).name,
-        })
+      outputs = {
+          PG_KEY: tf.squeeze(features[key_name]).name,
+          PG_TARGET: tf.squeeze(input_target_label).name,
+      }
 
-      # get top k answers
+      # TODO(brandondutra): get the score of the target label too.
+      #input_target_score = vector_slice(predictions, features[target_name])
+      
+      # get top k labels and their scores.
       (top_k_values, top_k_indices) = tf.nn.top_k(predictions, k=args.top_n)
       top_k_labels = tf.contrib.lookup.index_to_string(
           tf.to_int64(top_k_indices),
           mapping=string_value)
+   
+      # Write the top_k values using 2*top_k columns. 
+      num_digits = int(math.ceil(math.log(args.top_n, 10)))
```

Comment:
Can we force top_n < 100? The following logic seems complicated and hard to maintain. top_n >= 100 seems unreasonable already.